core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreReaderFragment.java

```diff
@@ -876,29 +879,54 @@ private void newTabInBackground(String url) {
   }
 
   private void closeTab(int index) {
-    tempForUndo = webViewList.get(index);
+    tempZimFileForUndo = zimReaderContainer.getZimFile();
+    tempWebViewForUndo = webViewList.get(index);
     webViewList.remove(index);
     if (index <= currentWebViewIndex && currentWebViewIndex > 0) {
       currentWebViewIndex--;
     }
     tabsAdapter.notifyItemRemoved(index);
     tabsAdapter.notifyDataSetChanged();
-    Snackbar.make(tabSwitcherRoot, R.string.tab_closed, Snackbar.LENGTH_LONG)
+    Snackbar undo = Snackbar.make(tabSwitcherRoot, R.string.tab_closed, Snackbar.LENGTH_LONG)
       .setAction(R.string.undo, v -> {
-        webViewList.add(index, tempForUndo);
-        tabsAdapter.notifyItemInserted(index);
-        tabsAdapter.notifyDataSetChanged();
-        Snackbar.make(snackbarRoot, "Tab restored", Snackbar.LENGTH_SHORT).show();
-        setUpWebViewWithTextToSpeech();
-      })
-      .show();
+        restoreDeletedTab(index);
+      });
+    undo.show();
+    //undo.addCallback(new Snackbar.Callback() {
```

Comment:
commented out code should be removed

Reply:
I wonder what a good method of remembering to delete 'test'/'debug' code is? I should probably skim the PR changes as well.

Reply:
Never a bad idea, I find the UI of sourcetree very beneficial to going through a commit before I actually send it through. Ideally when a PR is not a draft it should represent the minimal and correct implementation of a feature